### PR TITLE
adds forcing from GCM for full-depth profile scm runs

### PR DESCRIPTION
**Pull request by Julians42**
## Purpose 

Allows single column model to be run to the top of the atmosphere using forcing from a gcm file. 

## To-do

Need to generalize to any cfsite - maybe do this as command line arg? 
Make experiment more physical

## Content

with @szy21 and @costachris 

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: config/model_configs/prognostic_edmfx_gcmdriven_fullcolumn.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+config: "column"
+z_max: 40e3
+z_elem: 60
+z_stretch: true
+dz_bottom: 30
+dz_top: 3000
+perturb_initstate: false
+dt: "10secs"
+t_end: "48hours"
+dt_save_state_to_disk: "6hours"
```
Comment (costachris) [inline config/model_configs/prognostic_edmfx_gcmdriven_fullcolumn.yml:28]: may be helpful to add `ode_algo: ARS343` somewhere to ensure we're using the right solver when running with Revise

**Review by costachris — APPROVED** [review]

File: src/callbacks/callbacks.jl [hunk 2]
```diff
@@ -190,6 +190,13 @@ function set_insolation_variables!(Y, p, t, ::RCEMIPIIInsolation)
     rrtmgp_model.weighted_irradiance .= FT(551.58)
 end
 
+function set_insolation_variables!(Y, p, t, ::GCMDrivenInsolation)
+    FT = Spaces.undertype(axes(Y.c))
```
Comment (szy21) [inline src/callbacks/callbacks.jl:194]:
```suggestion
```

File: config/model_configs/prognostic_edmfx_gcmdriven_fullcolumn.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+initial_condition: "GCM"
+external_forcing: "GCM"
+external_forcing_file: "/central/groups/esm/zhaoyi/GCMForcedLES/forcing/corrected/HadGEM2-A_amip.2004-2008.07.nc"
```
[the remark this answers is not in the document]
Comment (Julians42) [reply config/model_configs/prognostic_edmfx_gcmdriven_fullcolumn.yml:3]: Good reminder! We are planning to add this in the next PR

File: src/callbacks/callbacks.jl (hunk 2 above)
Comment (Julians42) [reply src/callbacks/callbacks.jl:194]: deleted and updating in next commit

File: config/model_configs/prognostic_edmfx_gcmdriven_fullcolumn.yml (hunk 1 above)
Comment (Julians42) [reply config/model_configs/prognostic_edmfx_gcmdriven_fullcolumn.yml:28]: added a line to the yaml file for this (renamed from` _fullcolumn.yml` to `_column.yml`)

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 4]
```diff
@@ -56,39 +56,47 @@ function external_forcing_cache(Y, external_forcing::GCMForcing)
...
                 zc_gcm,
-                zc_les,
-                gcm_driven_profile_tmean(ds, varname),  # TODO: time-varying tendencies
+                zc_forcing,
+                gcm_driven_profile_tmean(ds.group["site23"], varname), 
             )
         end
 
-        function setnudgevar!(cc_field, varname, colidx, zc_gcm, zc_les)
+        insolation[1] = mean(ds.group["site23"]["rsdt"][:] ./ ds.group["site23"]["coszen"][:]) #similar(Fields.level(Y.c.ρ, 1), FT) #similar(Spaces.level(Y.c.ρ, 1), FT)
```
[the remark this answers is not in the document]
Comment (Julians42) [reply src/prognostic_equations/forcing/external_forcing.jl]: done

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 5]
```diff
@@ -145,8 +155,8 @@ function external_forcing_tendency!(Yₜ, Y, p, t, ::GCMForcing)
 
     ᶜdTdt_sum = p.scratch.ᶜtemp_scalar
     ᶜdqtdt_sum = p.scratch.ᶜtemp_scalar_2
-    @. ᶜdTdt_sum = ᶜdTdt_hadv + ᶜdTdt_fluc + ᶜdTdt_rad + ᶜdTdt_nudging
-    @. ᶜdqtdt_sum = ᶜdqtdt_hadv + ᶜdqtdt_fluc + ᶜdqtdt_nudging
+    @. ᶜdTdt_sum = ᶜdTdt_hadv + ᶜdTdt_nudging #  + ᶜdTdt_rad + ᶜdTdt_fluc remove nudging for now - TODO add back later
```
[the remark this answers is not in the document]
Comment (Julians42) [reply src/prognostic_equations/forcing/external_forcing.jl]: done

File: config/model_configs/prognostic_edmfx_gcmdriven_column.yml [hunk 6]
```diff
@@ -15,23 +16,34 @@ edmfx_nh_pressure: true
...
+z_stretch: true
+dz_bottom: 30
+dz_top: 3000
 perturb_initstate: false
 dt: "10secs"
 t_end: "6hours"
-dt_save_state_to_disk: "10mins"
+dt_save_state_to_disk: "6hours"
+call_cloud_diagnostics_per_stage : true
+output_dir: output/gcm_driven_scm
```
Comment (szy21) [inline config/model_configs/prognostic_edmfx_gcmdriven_column.yml:30]:
```suggestion
```

File: config/model_configs/prognostic_edmfx_gcmdriven_column.yml [hunk 7]
```diff
@@ -1,11 +1,12 @@
 initial_condition: "GCM"
 external_forcing: "GCM"
-external_forcing_file: "/groups/esm/zhaoyi/GCMForcedLES/cfsite/07/HadGEM2-A/amip/Output.cfsite23_HadGEM2-A_amip_2004-2008.07.4x/stats/Stats.cfsite23_HadGEM2-A_amip_2004-2008.07.nc"
+external_forcing_file: "/central/groups/esm/zhaoyi/GCMForcedLES/forcing/corrected/HadGEM2-A_amip.2004-2008.07.nc" 
 surface_setup: "GCM"
 turbconv: "prognostic_edmfx"
 implicit_diffusion: true
 implicit_sgs_advection: false
 approximate_linear_solve_iters: 2
+max_newton_iters_ode: 1
```
Comment (szy21) [inline config/model_configs/prognostic_edmfx_gcmdriven_column.yml:9]:
```suggestion
```

**Pull request merged**